### DISCUSSION TITLE
Issue building l4t/l4t-r35.3.ga-5.10 kernel version

### DISCUSSION THREAD

**Discussion by NavSriv** [OE4T/meta-tegra#1490] [Q&A] [answered]
I am using [Diamond Systems Jetson Orin NX 16GB](https://www.diamondsystems.com/products/jackson) board for development. It has 128GB SSD on it. I am building image through

poky -> kirkstone
meta-tegra -> kirkstone intermediate revision (because boards boots up with Jetpack BSP version -35.3.1, Kernel 5.10.104, checked-out revision dd0d91ad1ac0f0654aed6cce143ed8fc26719196 which support jetpack 5.1.1)

I have following change in my local.conf -

MACHINE ??= "p3509-a02-p3767-0000"
TNSPEC_BOOTDEV:p3509-a02-p3767-0000 = "nvme0n1p1"

I want to use "l4t/l4t-r35.3.ga-5.10" branch of linux-tegra, so I created bbappend for linux-tegra. Below is how bbappend looks like -

```c++
FILESEXTRAPATHS:prepend := "${THISDIR}/${PN}:"

SRCBRANCH = "l4t/l4t-r35.3.ga-5.10"
SRCREV = "78c2b512f23554067081db44a45ae59309bc6917"

SRC_REPO = "github.com/OE4T/linux-tegra-5.10.git;protocol=https"
KERNEL_REPO = "${SRC_REPO}"

SRC_URI = "git://${KERNEL_REPO};name=machine;branch=${KBRANCH} "
``` 

"do_kernel_configme" fails and log say it fails at

```
Using .kernel-meta/configs///defconfig as base
make[1]: Entering directory '/home/JetsonOrinNX/DiamondSystem/poky-bridge12/build/tmp/work/p3509_a02_p3767_0000-poky-linux/linux-tegra/5.10.104+gitAUTOINC+78c2b512f2-r0/linux-p3509_a02_p3767_0000-standard-build'
  GEN     Makefile
drivers/video/Kconfig:27: can't open file "drivers/video/tegra/Kconfig"
make[2]: *** [/home/JetsonOrinNX/DiamondSystem/poky-bridge12/build/tmp/work-shared/p3509-a02-p3767-0000/kernel-source/scripts/kconfig/Makefile:71: alldefconfig] Error 1
``` 

Appreciate any pointers. Thank you.

**Comment by kekiefer** [comment]
You probably should be using the `oe4t-patches-l4t-r35.3.ga` branch instead. The `l4t/l4t-r35.3.ga-5.10` reflects the upstream source for that release, but not the additions needed to make it work with the OE build system.

**Reply by NavSriv** [reply]
The source which had been provided along with the board is inline with l4t/l4t-r35.3.ga-5.10 so I was intended to change the branch. I will create patch for oe4t-patches-l4t-r35.3.ga and test. Thanks for the information.

**Reply by NavSriv** [reply]
This worked.. Board was up, I am assuming with logs but I am ending up in UEFI interactive shell -

![IMG_4473](https://github.com/OE4T/meta-tegra/assets/144927409/38ac3bab-68f3-4e77-8ead-f8c94b373808)

Does that mean rootfs is getting properly mounted? 
Thanks

**Reply by kekiefer** [reply, marked as answer]
I would say if it booted into the uefi shell instead of loading a kernel, then it can't find the kernel partition (or rootfs if it's trying to boot extlinux), which would be on external storage. Is the dtb you are using set up to enable the PCIe controller where your NVMe is attached?

**Reply by NavSriv** [reply]
yes, that is a possibility which I overlooked. I have Jetson pack changes for the board from the provider and trying to convert them into yocto changes. I hadn't made uphy lane change for PCIe in DTS and config file at the moment. That could be the issue. I will port all the changes and post the result here.
Thanks for your suggestion.

**Comment by NavSriv** [comment]
> I would say if it booted into the uefi shell instead of loading a kernel, then it can't find the kernel partition (or rootfs if it's trying to boot extlinux), which would be on external storage. Is the dtb you are using set up to enable the PCIe controller where your NVMe is attached?

Changed dtb and it worked. Thanks
